src/services/user.js

```diff
@@ -328,6 +328,36 @@ const userService = {
     const userCooperations = await cooperationService.getCooperations(aggregateOptions)
 
     return Boolean(userOffers?.length || userCooperations?.length)
+  },
+
+  checkOwnership: async (model, ownerFields, resourceId, userId, relationshipModel) => {
+    const resource = await model.findById(resourceId)
+    if (!resource) throw createError(404, DOCUMENT_NOT_FOUND([model.resourceType]))
+
+    const isOwner = ownerFields.some((field) => resource[field] && resource[field].toString() === userId)
+
+    if (isOwner) return resource
+
+    if (relationshipModel) {
+      const { sectionsResources, resourceField, userFields, availabilityField } = relationshipModel.dynamicPaths
+
+      const elemMatchQuery = { [resourceField]: resourceId }
+
+      if (availabilityField) {
+        elemMatchQuery[availabilityField] = 'open'
+      }

```

Comment:
Just a question.

If it is **ownership** check, why availability checking is here? When I think about ownership checking, first thing that came to my mind - check if the user has enough rights to get this resource. So you can check just only those fields, that related to **ownership**, but availability - it's just different.  
Probably you can split this middleware into 2. One will perform ownership checking, other - availability checking. I think it will simplify query for each middleware, and also will be more combined with single responsibility principle.

Again, it's just a question. You did a great job!